truffle/test/05_eth_api.js

```diff
@@ -15,4 +33,41 @@ describe('Ethereum API', () => {
       assert.equal(true, false, "Invalid selected network");
     }
   });
+
+  it("should assert whether deployment does not successfully complete using estimated gas", async () => {
+    const myContract = new _web3.eth.Contract(HelloBlockchainWorld.abi);
+    const estimatedGas = await myContract.deploy({
+      data: HelloBlockchainWorld.bytecode,
+      arguments: []
+    }).estimateGas();
+
+    const deployContract = () => {
+      return new Promise((resolve, reject) => {
+        myContract.deploy({
+          data: HelloBlockchainWorld.bytecode,
+          arguments: []
+        }).send({ gas: estimatedGas })
+            .on('error', (err) => {
+              reject(err)
+            })
+            .on('confirmation', (confirmationNumber, receipt) => {
+              resolve(receipt);
+            })
+      });
+    };
```

Comment:
can't we use ES6 syntax here?

Reply:
In this case, it is not possible unfortunally

Reply:
promises ftw